.github/workflows/ci.yml

```diff
@@ -0,0 +1,78 @@
+name: CI
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+  release:
+    types: [created]
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        scala_version: &scala_versions
+          - 2.12.12
+          - 2.13.4
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with: &cache_options
+          path: |
+            ~/.ivy2/cache
+            ~/.sbt
+            ~/.coursier
+          key: ${{ runner.os }}-sbt-${{ hashFiles('**/*.sbt') }}-${{ matrix.scala_version }}
+          restore-keys:
+            - ${{ runner.os }}-sbt-${{ hashFles('**/*.sbt') }}-
+      - name: Build
+        run: sbt ++${{ matrix.scala_version }} compile packageBin packageDoc packageSrc
+      - name: Test
+        run: sbt ++${{ matrix.scala_version }} test
+      - uses: actions/upload-artifact@v2
+        with:
+          name: target-${{ matrix.scala_version }}
+          path: '*/target/'
+          retention-days: 1
+  publish:
+    if: github.event_name == "push" || github.event_name == "release"
+    needs: build-and-test
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        scala_version: *scala_versions
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with: *cache_options
+      - uses: actions/download-artifact@v2
+        with:
+          name: target-${{ matrix.scala_version }}
+      - name: Publish to Sonatype
+        run: bash publish.sh
+        env:
+          SCALA_VERSION: ${{ matrix.scala_version }}
+          PGP_SECRET: ${{ secrets.PGP_SECRET }}
+  release-upload:
+    if: github.event_name == "release"
+    needs: build-and-test
+    strategy:
+      matrix:
+        scala_version: *scala_versions
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          name: target-${{ matrix.scala_version }}
+      - name: Upload assets
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          set -x
+          tag="${GITHUB_REF#refs/tags/}"
+          args=$(printf -- "-a %q " {xtract-core,testing,macros}/target/*/*.jar)
+          hub release edit $args -m "" $tag
```

Comment:
we weren't publishing `macros/target/*/*.jar` before; double checking this change intentional

Reply:
yep. It should have been published, but wasn't.